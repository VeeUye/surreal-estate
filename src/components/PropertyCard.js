import React from "react";
import { BiBuildingHouse } from "react-icons/bi";
// import HouseFront from "../assets/images/house-front.jpg";

// import PropTypes from "prop-types";
import "../styles/property-card.css";

const PropertyCard = () => {
  return (
    <div className="property-card">
      <div className="property-card__header">
        <BiBuildingHouse className="property-card__logo" />
      </div>
      <div className="property-card__details">
        <h2 className="property-card__details property-card__details__h">
          Title
        </h2>
        <p className="property-card__details property-card__details__p">
          Type - Location
        </p>
        <p className="property-card__details property-card__details__p">
          Baths
        </p>
        <p className="property-card__details property-card__details__p">Beds</p>
        <p className="property-card__details property-card__details__p">
          Price
        </p>
      </div>
      <button type="button" className="property-card__button">
        Email
      </button>

      {/* <img
        className="property-card property-card__img"
        src={HouseFront}
        alt="interior of bedroom 2"
      /> */}
    </div>
  );
};

export default PropertyCard;
