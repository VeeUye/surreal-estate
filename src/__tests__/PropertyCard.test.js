import React from "react";
import { getByAltText, render, screen } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("Property Card", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<PropertyCard />);

    expect(asFragment()).toMatchSnapshot();
  });

  xit("displays a property title", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/2 bed flat/i).textContent).toBe("2 bed flat");
  });

  xit("displays a property type", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/Flat/).textContent).toBe("Flat");
  });

  xit("displays the correct number of baths", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/bath/).textContent).toBe("2 bath");
  });

  xit("displays the correct number of beds", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/bed/).textContent).toBe("2 bed");
  });

  xit("displays the correct price", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/£/).textContent).toBe("£100000");
  });

  xit("displays the correct city", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/Manchester/).textContent).toBe("Manchester");
  });

  xit("displays the correct number of beds", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/bed/).textContent).toBe("2 bed");
  });

  xit("renders the email button", () => {
    render(<PropertyCard />);

    expect(screen.getByRole("button", { name: /email/ }));
  });

  it("displays an image", () => {
    render(<PropertyCard />);
    const image = screen.getByAltText(/interior of bedroom 2/i);

    expect(image).toBeVisible();
  });

  it("displays a logo", () => {
    render(<PropertyCard />);
    const image = screen.getByAltText(/Surreal Estate/i);

    expect(image).toBeVisible();
  });
});
