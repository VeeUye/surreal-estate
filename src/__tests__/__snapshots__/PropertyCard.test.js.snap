// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Card renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <img
      alt="interior of bedroom 2"
      class="property-card property-card__img"
      src="house-front.jpg"
    />
    <img
      alt="Surreal Estate"
      class="property-card property-card__se-logo"
      src="[object Object]"
    />
  </div>
</DocumentFragment>
`;
